### ISSUE TITLE
Optimize performance

### ISSUE THREAD

**Issue by d3lm** [d3lm/ngx-drag-to-select#14]
As I found, this library is already very fast and with two instances on the same page I get roughly 59 FPS on average. So rendering seems to be quite good. Yet it needs to be investigated where to use micro optimizations such as the `animationFrame` scheduler as discussed in #9.

Additionally, research is needed where to run things outside of the zone to reduce unnecessary change detection phases. As part of this, one could also look at where to use `ChangeDetectionStrategy.OnPush`.

**Comment by vitaliy-bobrov** [comment]
It is possible to use `{passive: true}` option for events like `resize`/`scroll` to improve performance - [info](https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md)

Event empty (`() => {}`) non-passive listener on `scroll` event creates page perf overhead.

**Comment by d3lm** [comment]
Yes indeed, I also looked into this once. This is definitely an optimization that can / should be added.

**Comment by d3lm** [comment]
@vitaliy-bobrov Mind sending in a PR?

**Comment by shyamal890** [comment]
Serious performance issues as list grows.

**Comment by shyamal890** [comment]
Setting `selectOnDrag` to `false` helped given there were some calculations happening on every select change which now only happen at `dragEnd`. I think by default `selectOnDrag` should be set to `false`.

**Comment by Savers80** [comment]
Hi All,
about performance i have setted [disabled]="true" or [selectOnDrag]="false" on element <dts-select-container> but without any click i see that container fire events: ngDoCheck, ngAfterContentChecked, ngAfterViewChecked simply moving the mouse ;
It is possible avoids this and fire this event ony when user start with selection area ?

thanks

**Comment by Savers80** [comment]
Hi d3lm,
about performance i have noted this:
Component <dts-select-container> is present and disabled in a component A and A is not rendered;
in a sibling component B i have logged this event ngDoCheck,ngAfterContentChecked, ngAfterViewChecked in this case no event is logged; 
the problem is when componet A is rendered ; in this case when i move the mouse without click events are logged evrywhere in page 

Please, How i can avoid this?

Thanks

**Comment by d3lm** [comment]
Thanks for bringing this up. Can you give me the StackBlitz link so I can look at your code?

**Comment by Savers80** [comment]
Hi,  
please,  this link to look scr ;
https://stackblitz.com/edit/ngx-drag-to-select-test

Thanks
